pkg/controller/kogitosupportingservice/kogitodataindex_controller.go

```
@@ -0,0 +1,266 @@
+				},
+			},
+		},
+		{Objects: []runtime.Object{&corev1.Service{}, &appsv1.Deployment{}, &corev1.ConfigMap{}}},
+	}
+	if err = controllerWatcher.Watch(watchedObjects...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileKogitoDataIndex implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileKogitoDataIndex{}
+
+// ReconcileKogitoDataIndex reconciles a KogitoSupportingService object
+type ReconcileKogitoDataIndex struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client *client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a KogitoSupportingService object and makes changes based on the state read
+// and what is in the KogitoSupportingService.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileKogitoDataIndex) Reconcile(request reconcile.Request) (result reconcile.Result, resultErr error) {
+	instance, resultErr := fetchKogitoSupportingService(r.client, request.Name, request.Namespace)
+	if resultErr != nil {
+		return
+	}
+	// Ignore reconcile of other services triggered when change in kogitoInfra object
+	if appv1alpha1.DataIndex != instance.Spec.ServiceType {
+		return
+	}
+	log.Infof("Reconciling KogitoDataIndex for %s in %s", request.Name, request.Namespace)
+	if resultErr = ensureSingletonService(r.client, request.Namespace, instance.Spec.ServiceType); resultErr != nil {
+		return
+	}
+
+	log.Infof("Injecting Data Index URL into KogitoRuntime services in the namespace '%s'", instance.Namespace)
+	if err := infrastructure.InjectDataIndexURLIntoKogitoRuntimeServices(r.client, instance.Namespace); err != nil {
```

This error seems to be swallowed, shouldn't it be rather:
```suggestion
	if resultErr := infrastructure.InjectDataIndexURLIntoKogitoRuntimeServices(r.client, instance.Namespace); resultErr != nil {
```

---

Ah yes, the error was swallowed, changed. Thanks :)